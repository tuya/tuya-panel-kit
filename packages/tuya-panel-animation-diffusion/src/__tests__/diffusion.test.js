/**
 * @jest-environment jsdom
 */
import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import Diffusion from '../index';

describe('Diffusion components', () => {
  it('basic render', () => {
    const wrapper = shallow(
      <Diffusion
        children={
          <View style={{ width: 100, height: 100, borderRadius: 50, backgroundColor: '#fff' }} />
        }
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('componentWillReceiveProps render', () => {
    jest.useFakeTimers();
    let startAnimated = false;
    const wrapper = shallow(
      <Diffusion
        startAnimated={startAnimated}
        children={
          <View style={{ width: 100, height: 100, borderRadius: 50, backgroundColor: '#fff' }} />
        }
      />
    );
    startAnimated = true;
    wrapper.setProps({ startAnimated });
    const instance = wrapper.instance();
    instance.timeHandle.push(jest.runOnlyPendingTimers());
    expect(wrapper).toMatchSnapshot();
    startAnimated = false;
    wrapper.setProps({ startAnimated });
    wrapper.unmount();
  });
});
