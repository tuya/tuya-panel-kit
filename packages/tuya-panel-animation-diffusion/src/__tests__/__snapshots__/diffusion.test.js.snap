// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diffusion components basic render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 400,
        "justifyContent": "center",
        "width": 400,
      },
      Object {},
    ]
  }
>
  <AnimatedComponent
    key="0"
    style={
      Array [
        Object {
          "borderColor": "#00FF00",
          "borderWidth": 10,
          "position": "absolute",
        },
        Object {
          "borderRadius": 100,
          "height": 200,
          "opacity": 0,
          "width": 200,
        },
      ]
    }
  />
  <AnimatedComponent
    key="1"
    style={
      Array [
        Object {
          "borderColor": "#00FF00",
          "borderWidth": 10,
          "position": "absolute",
        },
        Object {
          "borderRadius": 100,
          "height": 200,
          "opacity": 0,
          "width": 200,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 200,
        "justifyContent": "center",
        "width": 200,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 50,
          "height": 100,
          "width": 100,
        }
      }
    />
  </View>
</View>
`;

exports[`Diffusion components componentWillReceiveProps render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 400,
        "justifyContent": "center",
        "width": 400,
      },
      Object {},
    ]
  }
>
  <AnimatedComponent
    key="0"
    style={
      Array [
        Object {
          "borderColor": "#00FF00",
          "borderWidth": 10,
          "position": "absolute",
        },
        Object {
          "borderRadius": 100,
          "height": 200,
          "opacity": 1,
          "width": 200,
        },
      ]
    }
  />
  <AnimatedComponent
    key="1"
    style={
      Array [
        Object {
          "borderColor": "#00FF00",
          "borderWidth": 10,
          "position": "absolute",
        },
        Object {
          "borderRadius": 100,
          "height": 200,
          "opacity": 1,
          "width": 200,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 200,
        "justifyContent": "center",
        "width": 200,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 50,
          "height": 100,
          "width": 100,
        }
      }
    />
  </View>
</View>
`;
